bin/init.sh

```diff
@@ -44,51 +48,76 @@ if [ ${ROOT} != "${GO_TOP:-$HOME/go}/src/istio.io/istio" ]; then
        exit 1
 fi
 
-PROXYVERSION=$(grep envoy-debug pilot/docker/Dockerfile.proxy_debug  |cut -d: -f2)
-PROXY=debug-$PROXYVERSION
+# Download Envoy debug and release binaries for Linux x86_64. They will be included in the
+# docker images created by Dockerfile.proxy and Dockerfile.proxy_debug.
 
-# Save envoy in $out
-if [ ! -f $OUT/envoy-$PROXYVERSION ] ; then
-    mkdir -p $OUT
-    pushd $OUT
-    # New version of envoy downloaded. Save it to cache, and clean any old version.
-
-    DOWNLOAD_COMMAND=""
+# Gets the download command supported by the system (currently either curl or wget)
+DOWNLOAD_COMMAND=""
+set_download_command () {
+    # Try curl.
     if command -v curl > /dev/null; then
-       if curl --version | grep Protocols  | grep https > /dev/null; then
-	   DOWNLOAD_COMMAND='curl -fLSs'
-       else
-           echo curl does not support https, will try wget for downloading files.
-       fi
-    else
-       echo curl is not installed, will try wget for downloading files.
-    fi
-
-    if [ -z "${DOWNLOAD_COMMAND}" ]; then
-        if command -v wget > /dev/null; then
-	    DOWNLOAD_COMMAND='wget -qO -'
-        else
-            echo wget is not installed.
+        if curl --version | grep Protocols  | grep https > /dev/null; then
+	       DOWNLOAD_COMMAND='curl -fLSs'
+	       return
         fi
+        echo curl does not support https, will try wget for downloading files.
+    else
+        echo curl is not installed, will try wget for downloading files.
     fi
 
-    if [ -z "${DOWNLOAD_COMMAND}" ]; then
-        echo Error: curl is not installed or does not support https, wget is not installed. \
-             Cannot download envoy. Please install wget or add support of https to curl.
-        exit 1
+    # Try wget.
+    if command -v wget > /dev/null; then
+        DOWNLOAD_COMMAND='wget -qO -'
+        return
     fi
-
-    echo "Downloading envoy $PROXY using $DOWNLOAD_COMMAND"
-    time ${DOWNLOAD_COMMAND} https://storage.googleapis.com/istio-build/proxy/envoy-$PROXY.tar.gz | tar xz
-    cp usr/local/bin/envoy $OUT/envoy-$PROXYVERSION
+    echo wget is not installed.
+
+    echo Error: curl is not installed or does not support https, wget is not installed. \
+         Cannot download envoy. Please install wget or add support of https to curl.
+    exit 1
+}
+
+# Normally set by the Makefile.
+# Keep the default URLs up-to-date with the latest push from istio/proxy.
+export ISTIO_ENVOY_DIR=${OUT_DIR}
+export ISTIO_ENVOY_VERSION=${ISTIO_ENVOY_VERSION:-54077a0c8486b6db7b2fc5d10619b6813bb40767}
```

Comment:
move the sha to a different variable
+ISTIO_ENVOY_DEFAULT_VERSION="54077a0c8486b6db7b2fc5d10619b6813bb40767"
+export ISTIO_ENVOY_VERSION=${ISTIO_ENVOY_VERSION:-ISTIO_ENVOY__DEFAULT_VERSION}

makes it easier for the update bot

Reply:
This duplication in logic (both existing, and now expanded) is getting a bit onerous and is a pain to maintain.  I wouldn't mind if we changed init.sh so it complained and failed if it isn't run via make (though I also wouldn't mind getting rid of the file in its entirety).

Reply:
I would avoid bots editing this file. 

Can you source istio.VERSION or some other file that is already updated by the bot ?